contribs/application/src/test/java/org/matsim/application/prepare/counts/CreateCountsFromBAStDataTest.java

```diff
@@ -0,0 +1,183 @@
+package org.matsim.application.prepare.counts;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.counts.Count;
+import org.matsim.counts.Counts;
+import org.matsim.counts.CountsReaderMatsimV1;
+import org.matsim.counts.MatsimCountsReader;
+import org.matsim.examples.ExamplesUtils;
+import org.matsim.testcases.MatsimTestUtils;
+
+import java.util.Map;
+
+public class CreateCountsFromBAStDataTest {
+
+	@Rule
+	public MatsimTestUtils utils = new MatsimTestUtils();
+
+	String carOutput = "car-test-counts.xml.gz";
+	String freightOutput = "freight-test-counts.xml.gz";
+
+	String ignoredCounts = "ignored.csv";
+	String manualMatchedCounts = "manual.csv";
+	String wrongManualMatchedCounts = "wrong_manual.csv";
+
+	String network = ExamplesUtils.getTestScenarioURL("berlin") + "network.xml.gz";
+	String motorwayData = "2021_A_S_test_data.txt";
+	String primaryData = "2021_B_S_test_data.txt.gz";
+	String stationData = "Jawe2021_test_data.csv";
+	String shp = "Bezirke-Berlin-shp.zip";
+
+	String networkCrs = "EPSG:31468";
+	String shpCrs = "EPSG:3857";
+
+	@Test
+	public void testCreateCountsFromBAStData(){
+
+		String version = "normal-";
+		String car = utils.getOutputDirectory() + version + carOutput;
+		String freight = utils.getOutputDirectory() + version + freightOutput;
+
+		String[] args = new String[]{
+				"--station-data=" + utils.getPackageInputDirectory() + stationData,
+				"--network=" + network,
+				"--network-crs=" + networkCrs,
+				"--motorway-data=" + utils.getPackageInputDirectory() + motorwayData,
+				"--primary-data=" + utils.getPackageInputDirectory() + primaryData,
+				"--shp=" + utils.getPackageInputDirectory() + shp,
+				"--shp-crs=" + shpCrs,
+				"--year=2021",
+				"--car-output=" + car,
+				"--freight-output=" + freight
+		};
+
+		new CreateCountsFromBAStData().execute(args);
+
+		Counts<Link> counts = new Counts<>();
+		new MatsimCountsReader(counts).readFile(car);
+
+		int size = counts.getCounts().size();
+
+		Assert.assertTrue(size > 0);
```

Comment:
assertThat(size).isLargerThan(0) leads to more readable output if the test fails.